com.unity.render-pipelines.high-definition/Runtime/Water/WaterSurface.cs

```diff
@@ -0,0 +1,399 @@
+        public AnimationCurve windFoamCurve = new AnimationCurve(new Keyframe(0f, 0.0f), new Keyframe(0.2f, 0.0f), new Keyframe(0.3f, 1.0f), new Keyframe(1.0f, 1.0f));
+        #endregion
+
+        // Internal simulation data
+        internal WaterSimulationResources simulation = null;
+
+        internal bool CheckResources(CommandBuffer cmd, int bandResolution, int bandCount, ref bool initialAllocation)
+        {
+            bool needUpdate = false;
+            // If the resources have not been allocated for this water surface, allocate them
+            if (simulation == null)
+            {
+                simulation = new WaterSimulationResources();
+                simulation.AllocateSmmulationResources(bandResolution, bandCount);
+                needUpdate = true;
+                initialAllocation = true;
+            }
+            else if (!simulation.ValidResources(bandResolution, bandCount))
+            {
+                simulation.ReleaseSmmulationResources();
+                simulation.AllocateSmmulationResources(bandResolution, bandCount);
+                needUpdate = true;
+                initialAllocation = true;
+            }
+
+            if (simulation.windSpeed != windSpeed
+                || simulation.windOrientation != windOrientation
+                || simulation.windAffectCurrent != windAffectCurrent
+                || simulation.patchSizes.x != waterMaxPatchSize)
+            {
+                needUpdate = true;
+                initialAllocation = false;
+            }
+
+            // The simulation data are not valid, we need to re-evaluate the spectrum
+            if (needUpdate)
+            {
+                UpdateSimulationData();
+            }
+
+            return !needUpdate;
+        }
+        private void Start()
+        {
+            // Add this water surface to the internal surface management
+            RegisterInstance(this);
+        }
+
+        private void Awake()
+        {
+            // Add this water surface to the internal surface management
+            RegisterInstance(this);
+        }
+
+        void UpdateSimulationData()
+        {
+            simulation.windSpeed = windSpeed;
+            simulation.windOrientation = windOrientation;
+            simulation.windAffectCurrent = windAffectCurrent;
+            simulation.patchSizes = HDRenderPipeline.ComputeBandPatchSizes(waterMaxPatchSize);
+            simulation.patchWindSpeed = HDRenderPipeline.ComputeWindSpeeds(simulation.windSpeed, simulation.patchSizes);
+        }
+
+        void OnDestroy()
+        {
+            // Remove this water surface from the internal surface management
+            UnregisterInstance(this);
+
+            // Make sure to release the resources if they have been created (before HDRP destroys them)
+            if (simulation != null && simulation.AllocatedTextures())
+                simulation.ReleaseSmmulationResources();
```

Comment:
I think this is a typo, it should be `ReleaseSimulationResources`

Note: Sorry if I shouldn't comment on the PR, I saw the comment you left a while back about this but I still thought I should point this out. Have a great day.

Reply:
https://media.giphy.com/media/BIZkwFtu2xDlS/giphy.gif